### ISSUE TITLE
Unable to install latest version of FFTW on Julia 1.6

### ISSUE THREAD

**Issue by bgroenks96** [JuliaMath/FFTW.jl#206]
Error during precompilation:

```
ERROR: LoadError: AssertionError: precompile(Tuple{typeof(rfft_output_size), Tuple{Int, Int, Int}, D})
Stacktrace:
 [1] _precompile_()
   @ FFTW ~/.julia/packages/FFTW/Iu2GG/src/precompile.jl:8
 [2] top-level scope
   @ ~/.julia/packages/FFTW/Iu2GG/src/FFTW.jl:65
 [3] include
   @ ./Base.jl:386 [inlined]
 [4] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
   @ Base ./loading.jl:1213
 [5] top-level scope
   @ none:1
 [6] eval
   @ ./boot.jl:360 [inlined]
 [7] eval(x::Expr)
   @ Base.MainInclude ./client.jl:446
 [8] top-level scope
```

Version:
```
      Status `~/.julia/environments/v1.6/Project.toml`
  [7a1cc6ca] FFTW v1.4.1
```

**Comment by bgroenks96** [comment]
The problem is resolved by downgrading to version 1.3.1:

```
   Resolving package versions...
    Updating `~/.julia/environments/v1.6/Project.toml`
  [7a1cc6ca] ↓ FFTW v1.4.1 ⇒ v1.3.1
    Updating `~/.julia/environments/v1.6/Manifest.toml`
  [d360d2e6] ↑ ChainRulesCore v0.9.45 ⇒ v0.10.2
  [31c24e10] ↑ Distributions v0.24.18 ⇒ v0.25.2
  [7a1cc6ca] ↓ FFTW v1.4.1 ⇒ v1.3.1
  [856f044c] ↓ MKL_jll v2021.1.1+1 ⇒ v2020.2.254+0

  Progress [========================================>]  2/2
2 dependencies successfully precompiled in 15 seconds (173 already precompiled)
```

**Comment by stevengj** [comment]
@bgroenks96, what is your `versioninfo()` output?

**Comment by ararslan** [comment]
No idea on my end why those assertions would be hit. Works fine for me with FFTW v1.4.1 on Linux with Julia v1.6.1.

**Comment by bgroenks96** [comment]
@stevengj 

```
Julia Version 1.6.0
Commit f9720dc2eb (2021-03-24 12:55 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Xeon(R) Gold 6154 CPU @ 3.00GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake-avx512)
Environment:
  JULIA_EDITOR = "/home/pd/bgroenke/.vscode-server/bin/cfa2e218100323074ac1948c885448fdf4de2a7f/node"
  JULIA_NUM_THREADS = 12
```

**Comment by bgroenks96** [comment]
The current version of FFTW works on my laptop, this problem seems to only occur on my institute's server. Perhaps they broke something...? If we could narrow down what possible system libraries could cause this, I can just yell at them to fix it :)

**Comment by giordano** [comment]
@bgroenks96 can you please try to start julia with
```
JULIA_PKG_SERVER="pkg.julialang.org" JULIA_DEPOT_PATH=$(mktemp -d) julia
```
and see if you still have the same problem precompiling `FFTW.jl` there?

**Comment by bgroenks96** [comment]
@giordano Yes, that worked! Could you explain what that changed? I had already tried deleting the entire `.julia` directory and that didn't fix the issue.

**Comment by giordano** [comment]
Well, that's equivalent to deleting the `.julia` directory: `JULIA_DEPOT_PATH` selects a different depot, without wiping out your `.julia`, which is something I do _not_ recommend in general.

**Comment by bgroenks96** [comment]
That's weird. There must be something else I installed in my environment that broke the FFTW compilation.

**Comment by KristofferC** [comment]
Ref https://github.com/JuliaMath/FFTW.jl/pull/214

**Comment by KristofferC** [comment]
Should be fixed via #214
